DirectXTex/DirectXTexSwizzle.cpp

```diff
@@ -0,0 +1,755 @@
+//-------------------------------------------------------------------------------------
+// DirectXTexSwizzle.cpp
+//
+// DirectX Texture Library - Standard Swizzle (z-order curve)
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+// http://go.microsoft.com/fwlink/?LinkId=248926
+//-------------------------------------------------------------------------------------
+
+#include "DirectXTex.h"
+#include "DirectXTexP.h"
+
+using namespace DirectX;
+
+
+namespace
+{
+#ifdef __AVX2__
+#define deposit_bits(v,m) _pdep_u32(v,m)
+#define extract_bits(v,m) _pext_u32(v,m)
+#else
+    // For ARM64 we could use SVE2-BITPERM intrinsics BDEP/BEXT if supported by the platform/compiler.
+
+    // N3864 - A constexpr bitwise operations library for C++
+    // https://github.com/fmatthew5876/stdcxx-bitops
+    uint32_t deposit_bits(uint32_t val, int mask) noexcept
+    {
+        uint32_t res = 0;
+        for (uint32_t bb = 1; mask != 0; bb += bb)
+        {
+            if (val & bb)
+            {
+                res |= mask & (-mask);
+            }
+            mask &= (mask - 1);
+        }
+        return res;
+    }
+
+    uint32_t extract_bits(uint32_t val, int mask) noexcept
+    {
+        uint32_t res = 0;
+        for (uint32_t bb = 1; mask !=0; bb += bb)
+        {
+            if (val & mask & -mask)
+            {
+                res |= bb;
+            }
+            mask &= (mask - 1);
+        }
+        return res;
+    }
+#endif
+
+    constexpr size_t MAX_TEXTURE_DIMENSION = 16384u;
+
+#if defined(_M_X64) || defined(_M_ARM64) || __x86_64__ || __aarch64__
+    constexpr uint64_t MAX_TEXTURE_SIZE = UINT64_C(16384) * UINT64_C(16384) * 16u;
+#else
+    constexpr uint64_t MAX_TEXTURE_SIZE = UINT32_MAX;
+#endif
+
+    // Standard Swizzle is not defined for these formats.
```

Comment:
Q: I'm curious where this list came from, from documentation or trial-and-error calling the API? Is `DXGI_FORMAT_R8_SNORM` valid, given it's not in the `switch`? I see mention in [the documentation](https://learn.microsoft.com/en-us/windows/win32/api/d3d12/ne-d3d12-d3d12_texture_layout) that `DXGI_FORMAT_R32G32B32_TYPELESS` is excluded (along with MSAA, 1D, depth and stencil), but otherwise I'm not seeing any particular list.